dataplane/globalbase.h

```diff
@@ -186,19 +186,36 @@ class generation
 	eResult tsc_state_update(const common::idp::updateGlobalBase::tsc_state_update::request& request);
 	eResult tscs_base_value_update(const common::idp::updateGlobalBase::tscs_base_value_update::request& request);
 
-	balancer_real_id_t* evaluate_service_ring_one_wrr(balancer_real_id_t* start,
-	                                   const balancer_real_id_t* const do_not_exceed,
-	                                   const balancer_service_t& service,
-	                                   balancer_service_range_t& range);
-	balancer_real_id_t* evaluate_service_ring_one_chash(balancer_real_id_t* start,
-	                                   const balancer_real_id_t* const do_not_exceed,
-	                                   const balancer_service_t& service,
-	                                   balancer_service_range_t& range);
-	balancer_real_id_t* evaluate_service_ring_one(balancer_real_id_t* start,
-	                                   const balancer_real_id_t* const do_not_exceed,
-	                                   const balancer_service_t& service,
-	                                   balancer_service_range_t& range);
-	void evaluate_service_ring();
+	enum class ServiceRingOp
+	{
+		Update,
+		Rebuild
+	};
+
+	std::vector<std::pair<balancer_real_id_t, decltype(balancer_real_state_t::weight)>>
```

Comment:
We use that long type at least three times, here, and two times in definition.
I suggest to add a `using` alias, will make it more readable.